design/crypto_primitives.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2018 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under the MIT license <LICENSE-MIT
+// https://opensource.org/licenses/MIT> or the Modified BSD license <LICENSE-BSD
+// https://opensource.org/licenses/BSD-3-Clause>, at your option. This file may not be copied,
+// modified, or distributed except according to those terms. Please review the Licences for the
+// specific language governing permissions and limitations relating to use of the SAFE Network
+// Software.
+
+//! Different MaidSafe tools need different crypto primitives. E.g. Crust only uses encryption
+//! public/private keys, Routing uses encryption and signing public/private keys, etc.
+//! Currently in some cases safe_crypto is unusable cause it doesn't expose some internal data like
+//! private keys, etc.
+//! One way to solve this and stay super flexible is to have 1-to-1 mappings with lower level
+//! crypto types from rust_sodium, miscreant, etc.
+//! This document attempts to catch safe_crypto design that would work for all our our cases.
+//!
+//! `PublicKeys` and `SecretKeys` currently contain both signing and encryption keys. In some
+//! cases, we don't need signing key at all, e.g. in Crust. So this document proposes to split
+//! `PublicKeys` and `SecretKeys` into corresponding types: `PublicEncryptKey, `SecretEncryptKey`,
+//! `PublicSignKey` and `SecretSignKey`.
+
+// no signing key
+pub struct PublicEncryptKey {
```

Comment:
for sake of completeness can we mention what traits it needs implemented (either via deriving or manual roll out) ?
Same for other types where applicable